### ISSUE TITLE
Unable to flash firmware

### ISSUE THREAD

**Issue by ways** [opensolarproject/OSPController#30]
Hi

I really hope this project isn't dead.

> flash with esptool.py: esptool.py --chip esp32 --port /dev/ttyUSB0 write_flash 0

This will flash successfully, but monitoring the output only gives:

> rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
> invalid header: 0x04030000
> invalid header: 0x04030000
> invalid header: 0x04030000
> invalid header: 0x04030000
> invalid header: 0x04030000
> invalid header: 0x04030000
> invalid header: 0x04030000
> ets Jul 29 2019 12:21:46

(repeats)

The nodemcu-32 is brand new.

**Comment by ways** [comment]
The idea seems so good I might rewrite this in esphome or micropython since I can't get this one to work.

**Comment by Tobiasonline** [comment]
try compile the sourcecode with vscode and platformio and flash your own binary. That works for me

**Comment by hyfinn** [comment]
> The idea seems so good I might rewrite this in esphome or micropython since I can't get this one to work.

Hey @ways - did you ever get round to rewriting? I'd be interested in any work you've done on this!

**Comment by ways** [comment]
> > The idea seems so good I might rewrite this in esphome or micropython since I can't get this one to work.
> 
> Hey @ways - did you ever get round to rewriting? I'd be interested in any work you've done on this!

Yes, I have it in a private repo. Will push to gitlab and post link within a few days.

**Comment by hyfinn** [comment]
> > > The idea seems so good I might rewrite this in esphome or micropython since I can't get this one to work.
> > 
> > 
> > Hey @ways - did you ever get round to rewriting? I'd be interested in any work you've done on this!
> 
> Yes, I have it in a private repo. Will push to gitlab and post link within a few days.

Amazing, thank you very much. I look forward to it!

**Comment by ways** [comment]
> > > > The idea seems so good I might rewrite this in esphome or micropython since I can't get this one to work.
> > > 
> > > 
> > > Hey @ways - did you ever get round to rewriting? I'd be interested in any work you've done on this!
> > 
> > 
> > Yes, I have it in a private repo. Will push to gitlab and post link within a few days.
> 
> Amazing, thank you very much. I look forward to it!

https://gitlab.com/larsfp/esphome-mppt

**Comment by Svyatoslavx** [comment]
Не работает и gitlab
